migration-engine/connectors/migration-connector/src/error.rs

```diff
@@ -35,14 +36,18 @@ impl ConnectorError {
 
     pub fn into_migration_failed(self, migration_name: String) -> Self {
         let context = self.context.clone();
-        let user_facing_error = self.user_facing_error.clone();
+        let user_facing_error = user_facing_errors::migration_engine::MigrationDoesNotApplyCleanly {
```

Comment:
The function name sounds generic while this error name sounds very specific. Should the function name change to reflect that? Seems to be a leftover of the error kind that was previously used.

Reply:
Good point, I'll rename to `into_migration_does_not_apply_cleanly()` and document it.